Tests/SentryTests/SentryCrash/SentryDebugMetaBuilderTests.swift

```diff
@@ -0,0 +1,152 @@
+@testable import Sentry
+import XCTest
+
+/**
+ * Some of the test parameters are copied from debugging
+ * SentryCrashReportConverter.convertDebugMeta.
+ */
+class SentryDebugMetaBuilderTests: XCTestCase {
+    
+    private var imageProvider: TestSentryCrashBinaryImageProvider!
+    private var sut: SentryDebugMetaBuilder!
```

Comment:
M: if you wrap in a `Fixture` class you can get `TestSentryCrashBinaryImageProvider` as a param, so if for some reason you need 2 different results for your fake, you are able to pass it.
right now you'd need to create a fresh new `SentryDebugMetaBuilder` from a specific test affecting test performance because of initing things on `setUp` that are not used in all the tests.

Reply:
I think it's the proper way to create a fresh `SentryDebugMetaBuilder` for each test, because I want the tests to be deterministic. I don't think creating a fresh `SentryDebugMetaBuilder` for each test has a real measurable impact on performance. I use all the things I set up in `setup` in every test. Don't get me wrong. The fixture pattern makes sense, but if the setup is super simple I don't get the advantage. It makes the code more complicated to read IMO.

Reply:
guess you got it wrong or I could not express myself well.
you do need to create a fresh `SentryDebugMetaBuilder` instance each test, with or without Fixture.

right now `TestSentryCrashBinaryImageProvider` does the job, but let's say that now you need a fake that throws an Error. The `TestSentryCrashBinaryImageProvider` instance on `setUp` doesn't work anymore.

You'd need to create a new instance of `TestSentryCrashBinaryImageProvider` that throws an error, but the `setUp` method already created one, so you have 2, but you'll use only one.

Maybe you don't need to test an error for `TestSentryCrashBinaryImageProvider`, but if you do, you'll have a performance impact, which can be nearly nothing, or a lot, if all the tests do this way.

Feel free to close it if it's not the case for this test, but I'd like to see this Fixture pattern forward if we need to test a class with Fakes that could return more than 1 state.